### PR TITLE
refactor(core): Unify workflow controllers (no-changelog)

### PR DESCRIPTION
**Pull request by ivov**
Combine EE workflows controller into main workflows controller, protecting paid functionality behind feature flag checks.

### CODE REVIEW

**Review by krynble — DISMISSED** [review]

File: packages/cli/src/workflows/workflows.controller.ts [hunk 1]
```diff
@@ -62,6 +66,24 @@ workflowsController.post(
 
 		WorkflowHelpers.addNodeIds(newWorkflow);
 
+		if (isSharingEnabled()) {
```
Comment (netroy) [inline packages/cli/src/workflows/workflows.controller.ts:69]: I kinda liked that we had two handlers for the same route, and we routed the requested correctly via a middleware. 
We don't need to change this now, but once we convert this router into a decorated class, I'd like to split these back up again into separate methods.

**Review by netroy — COMMENTED** [review]
besides the one comment, this is looking good.
but, I'll wait for @krynble to approve again.
Thanks for taking care of this 🙏🏽

File: packages/cli/src/workflows/workflows.controller.ts (hunk 1 above)
Comment (ivov) [reply packages/cli/src/workflows/workflows.controller.ts:69]: Yes, definitely agree. This is an interim state.

**Review by krynble — APPROVED** [review]

**Comment by Joffcom** [comment]
Got released with [`n8n@1.23.0`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.23.0)

**Pull request merged**